### PR TITLE
Fix scan-build reports

### PR DESCRIPTION
**Pull request by jimying**
Use clang scan-build (https://github.com/pjsip/pjproject/pull/3443 )  to analysis code  and fix the reports.

Here simple shell script to show how to:
```
JOBS=8
# exclude third_party
SCAN_BUILD="scan-build -o  output --exclude third_party"

export CC=clang
export CXX=clang++
export CFLAGS="-g -fPIC -Wextra"


make distclean

./configure .....

make dep

$SCAN_BUILD make -j  $JOBS
```

### CODE REVIEW

**Comment by sauwming** [comment]
Unfortunately it still fails some of the unit tests.

**Comment by jimying** [comment]
> Unfortunately it still fails some of the unit tests.

Revert a error modify,  test passed


Current state:
   
    All Bugs: 170  (error:  5,  Unused code: 165 )





- >appendix: errors left on linux:
```
Bug Group	Bug Type 	File	  Function/Method	  Line
API	Argument with 'nonnull' attribute passed null	pjlib/include/pj/string.h	pj_memcpy	797
Logic error	Division by zero	pjmedia/src/pjmedia/conference.c	get_frame	2103
Logic error	Division by zero	pjmedia/src/pjmedia/conference.c	get_frame	2189
Logic error	Division by zero	pjmedia/src/pjmedia-videodev/sdl_dev.c	job_thread	1408
Logic error	Dereference of null pointer	pnath/src/pjnath/ice_strans.c	use_buffer	1783
```

File: pjmedia/src/pjmedia/vid_port.c [hunk 1]
```diff
@@ -1143,7 +1147,7 @@ static pj_status_t get_frame_from_buffer(pjmedia_vid_port *vp,
     pj_mutex_lock(vp->frm_mutex);
     if (vp->conv.conv)
         status = convert_frame(vp, vp->frm_buf, frame);
-    else
+    else if (frame->buf)
```
Comment (sauwming) [inline pjmedia/src/pjmedia/vid_port.c:1150]: why is this necessary?
`pjmedia_frame_copy()` will perform checking on the destination frame size.

Comment (jimying) [reply pjmedia/src/pjmedia/vid_port.c:1150]: I revert this change 
pj_memcpy(NULL,  src, 0)  no crash

File: pjsip/src/pjsua-lib/pjsua_media.c [hunk 2]
```diff
@@ -3612,6 +3616,8 @@ static pj_bool_t is_media_changed(const pjsua_call *call,
         }
 
         /* Compare codec param */
+        if (new_cp == NULL)
```
Comment (sauwming) [inline pjsip/src/pjsua-lib/pjsua_media.c:3619]: if `pjmedia_stream_info_from_sdp()` returns success, I believe this cannot be NULL.

File: pjsip/src/pjsua-lib/pjsua_media.c [hunk 3]
```diff
@@ -3467,6 +3467,10 @@ static void check_srtp_roc(pjsua_call *call,
             new_rem_addr = &new_si_->info.vid.rem_addr;
         }
 #endif
+        else {
```
Comment (sauwming) [inline pjsip/src/pjsua-lib/pjsua_media.c:3470]: `check_srtp_roc()` is only called from `pjsua_media_channel_update()` when call_med->type is audio (surprisingly the video counterpart checking is missing! so this is actually the bug)

File: pjsip-apps/src/pjsua/pjsua_app_legacy.c [hunk 4]
```diff
@@ -318,7 +318,7 @@ static void vid_handle_menu(char *menuin)
         argv[argc] = strtok(NULL, " \t\r\n");
     }
 
-    if (argc == 1 || strcmp(argv[1], "help")==0) {
+    if (argc <= 1 || strcmp(argv[1], "help")==0) {
```
Comment (sauwming) [inline pjsip-apps/src/pjsua/pjsua_app_legacy.c:321]: since the command requires space, i.e. `vid `, argc can't be 0.
```
            if (menuin[1]=='i' && menuin[2]=='d' && menuin[3]==' ') {
                vid_handle_menu(menuin);
```

File: pjmedia/src/pjmedia/stream.c [hunk 5]
```diff
@@ -773,11 +773,11 @@ static pj_status_t get_frame( pjmedia_port *port, pjmedia_frame *frame)
                          "codec decode() error"));
 
                 if (use_dec_buf) {
-                    pjmedia_zero_samples(p_out_samp + samples_count,
-                                         samples_per_frame);
-                } else {
                     pjmedia_zero_samples(stream->dec_buf,
                                          stream->dec_buf_count);
+                } else {
```
Comment (sauwming) [inline pjmedia/src/pjmedia/stream.c:778]: wow, I can't believe this bug is undetected for so long.

File: pjmedia/src/pjmedia/rtp.c [hunk 6]
```diff
@@ -202,12 +202,12 @@ PJ_DEF(pj_status_t) pjmedia_rtp_decode_rtp2(
     }
 
     /* Check that offset is not greater than packet size */
-    if (offset > pkt_len) {
+    if (offset >= pkt_len) {
```
Comment (sauwming) [inline pjmedia/src/pjmedia/rtp.c:205]: This keeps going back and forth between `>` and `>=`
Please check https://github.com/pjsip/pjproject/pull/3283
I think the current version is already correct

**Review by sauwming — COMMENTED** [review]
I've checked everything else and they look okay to me.

File: pjsip/src/pjsua-lib/pjsua_media.c (hunk 2 above)
Comment (jimying) [reply pjsip/src/pjsua-lib/pjsua_media.c:3619]: better change to `PJ_ASSERT_RETURN(new_cp, PJ_FALSE);` ?

Comment (sauwming) [reply pjsip/src/pjsua-lib/pjsua_media.c:3619]: ok

File: pjsip/src/pjsua-lib/pjsua_media.c (hunk 3 above)
Comment (sauwming) [reply pjsip/src/pjsua-lib/pjsua_media.c:3470]: I fixed the missing video SRTP roc checking in #3479

File: pjlib-util/src/pjlib-util/resolver.c [hunk 7]
```diff
@@ -1703,7 +1703,7 @@ static void on_read_complete(pj_ioqueue_key_t *key,
                                  sizeof(resolver->tmp_pool));
 
     /* Parse DNS response */
-    status = -1;
+    // status = -1;
```
Comment (bennylp) [inline pjlib-util/src/pjlib-util/resolver.c:1706]: Just delete I think

File: pjmedia/src/pjmedia/conference.c [hunk 8]
```diff
@@ -551,6 +551,7 @@ PJ_DEF(pj_status_t) pjmedia_conf_create( pj_pool_t *pool,
     const pj_str_t name = { "Conf", 4 };
     pj_status_t status;
 
+    PJ_ASSERT_RETURN(samples_per_frame > 0, PJ_EINVAL);
```
Comment (bennylp) [inline pjmedia/src/pjmedia/conference.c:554]: If we're adding checking, I'm curious why only samples_per_frame is checked

**Review by bennylp — CHANGES_REQUESTED** [review]
Thanks for the PR. I see that some bugs were caught!

File: pjmedia/src/pjmedia/conference.c (hunk 8 above)
Comment (jimying) [reply pjmedia/src/pjmedia/conference.c:554]: a error "Division by zero" report in this line:
https://github.com/pjsip/pjproject/blob/2c4bf48fb5891a3bb1fff90932f40d69d94b8d75/pjmedia/src/pjmedia/conference.c#L2102

`conf->samples_per_frame`  assign in pjmedia_conf_create()

Add the assert  just to emphasize that this will not happen

**Review by sauwming — APPROVED** [review]

**Review by bennylp — APPROVED** [review]
